packages/connect/src/utils/firmwareReleaseConfigUtils.ts

```diff
@@ -6,122 +6,119 @@ import { getJWSPublicKey } from '@trezor/env-utils';
-        // When using local we always want to use CodeSignKey.
-        ['test-signed', 'production'].includes(env) || !isRemote,
-    );
-    if (!JWSPublicKey) {
-        throw new Error('JWS public key is not defined!');
+    if (!verify(jws, JWS_CONFIG.SIGN_ALGORITHM, publicKey)) {
+        throw new Error('JWS signature is invalid.');
     }
+};
 
-    try {
-        const isAuthenticityValid = verify(releasesJws, JWS_SIGN_ALGORITHM, JWSPublicKey);
-        if (!isAuthenticityValid) {
-            throw new Error('Config authenticity is invalid');
+export const getFirmwareReleaseConfig = async () => {
+    const { jws: remoteJws, source: initialSource, env } = await fetchRemoteJWS();
+
+    let finalJws = remoteJws;
+    let finalSource = initialSource;
+
+    if (initialSource === 'remote') {
+        try {
+            const remotePayload = decodeJwsPayload(remoteJws);
+            const localPayload = decodeJwsPayload(firmwareReleaseConfigAssets.jws);
+
+            // Sanity check local config version.
+            if (localPayload.version !== JWS_CONFIG.VERSION) {
+                throw new Error(`Local config has mismatched version: ${localPayload.version}`);
+            }
+
+            // If local is newer or same sequence, prefer it over remote.
+            if (localPayload.sequence >= remotePayload.sequence) {
+                finalJws = firmwareReleaseConfigAssets.jws;
+                finalSource = 'local';
+            }
+        } catch (error) {
+            console.error(`Error comparing remote/local JWS: ${error}. Using remote as is.`);
         }
-
-        return JSON.parse(decodedJws.payload);
-    } catch (error) {
-        console.error('Error validating:', error);
-        throw new Error(`Failed to validate release message: ${error.message}`);
     }
-};
 
-export const getFirmwareReleaseConfig = async () => {
-    const { releasesJws, isRemote, env } = await getReleaseJWS();
-    const config: FirmwareReleaseConfig = verifyFirmwareRelease(releasesJws, isRemote, env);
+    const useProductionKey = ['test-signed', 'production'].includes(env) || finalSource === 'local';
+    const publicKey = getJWSPublicKey('firmware-release', useProductionKey);
+
+    verifyJwsSignature(finalJws, publicKey);
+    const config = decodeJwsPayload(finalJws);
 
     return {
         config,
-        isRemote,
+        isRemote: finalSource === 'remote',
     };
 };
 
 export const getOnlyLocalFirmwareReleaseConfig = (): {
     config: FirmwareReleaseConfig;
     isRemote: false;
 } => {
-    // The bundled local is always the production one so hard-code production env here.
-    const config: FirmwareReleaseConfig = verifyFirmwareRelease(
-        firmwareReleaseConfigAssets.jws,
-        false,
-        'production',
-    );
+    const localJws = firmwareReleaseConfigAssets.jws;
+    // For local-only, we always use the production signing key.
+    const publicKey = getJWSPublicKey('firmware-release', true);
```

Comment:
Would be great to unify naming of this one as well:
```suggestion
    const publicKey = getJwsPublicKey('firmware-release', true);
```